pkg/lib/cockpit.d.ts

```diff
@@ -131,8 +131,17 @@ declare module 'cockpit' {
         close(): void;
     }
 
+    type VariantType = string | string[] | Uint8Array | number | number[] | boolean | boolean[];
+    interface Variant {
+        v: VariantType
+        t: string;
```

Comment:
I tried to see if we can do better than string, and I think we can't :(

Reply:
so can't we throw a loop in there? Or define Dbus types like `u`  `b` and then somehow add `a{s}` although one can repeat that right like uhhh `usb`

Reply:
I tried to do something like

```typescript
type DBusType = "b" | "i" | "s" | `a${DBusType}`;
```

but it complained about the self-reference.  Apparently that's only OK when building simple recursive types.  And even if that worked, I'm not sure how we could describe tuples, without getting into some weirdness like:

```typescript
type TupleType = `(${OneOrMoreDBusTypes})`;
type OneOrMoreDBusTypes = `${DBusType}` | `${OneOrMoreDBusTypes}${DBusType}`
```

...which really seems a bridge too far.